React-Projects/job-portal/src/Signup.jsx

```diff
@@ -1,62 +1,54 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./styles/Signup.css";
 
 function Signup() {
-  //   useEffect(() => {
-  //     document
-  //       .getElementById("registrationForm")
-  //       .addEventListener("submit", (event) => {
-  //         event.preventDefault();
-  //         var fullname = document.getElementById("fullname").value;
-  //         var email = document.getElementById("email").value;
-  //         var password = document.getElementById("password").value;
-  //         var confirm_password =
-  //           document.getElementById("confirm_password").value;
+  const [fullname, setFullname] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirm_password, setConfirmPassword] = useState("");
 
-  //         if (password !== confirm_password) {
-  //           alert("Passwords do not match");
-  //           return;
-  //         }
+  const registrationForm = (event) => {
+    event.preventDefault();
 
-  //         var requestBody = {
-  //           name: fullname,
-  //           email: email,
-  //           password: password,
-  //         };
+    if (password !== confirm_password) {
+      alert("Passwords do not match");
+      return;
+    }
 
-  //         fetch("http://127.0.0.1:8000/api/register", {
-  //           method: "POST",
-  //           headers: {
-  //             "Content-Type": "application/json",
-  //           },
-  //           body: JSON.stringify(requestBody),
-  //         })
-  //           .then(async (response) => {
-  //             if (!response.ok) {
-  //               const data = await response.json();
+    var requestBody = {
+      name: fullname,
+      email: email,
+      password: password,
+    };
 
-  //               if (data.email) {
-  //                 throw new Error(data.email[0]);
-  //               } else if (data.password) {
-  //                 throw new Error(data.password[0]);
-  //               } else {
-  //                 throw new Error("Network response is not ok");
-  //               }
-  //             }
-  //             return response.json();
-  //           })
-  //           .then((data) => {
-  //             alert("Registration successful");
-  //           })
-  //           .catch((error) => {
-  //             alert(`${error}. Please try again.`);
-  //           });
-  //       });
+    fetch("http://127.0.0.1:8000/api/register", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(requestBody),
+    })
+      .then(async (response) => {
+        if (!response.ok) {
+          const data = await response.json();
 
-  //     return () => {
-  //       second;
-  //     };
-  //   }, [third]);
+          if (data.email) {
```

Comment:
@codal-kcolaco not a good way to handle errors.. this will result in many errors.

Reply:
@abhi-codal ok I will check how to handle it better